blaze/utils.py

```diff
@@ -187,6 +192,11 @@ def json_dumps(ds):
     return {'__!datashape': str(ds)}
 
 
+@dispatch(types.BuiltinFunctionType)
```

Comment:
This dispatch is going to get hit for any function defined in C. This means that thinks like `np.sum` will get deserialized as `builtins.sum`. I think we might need some stronger checks here